samples/google/gmail/program.py

```diff
@@ -9,144 +9,134 @@
-slack = slack_client("slack_conn")
 
 
-def on_slack_slash_command(event):
-    """Use a Slack slash command to interact with a Gmail mailbox.
+def on_http_get(event):
+    """Handle Gmail interaction via HTTP trigger using query params.
 
-    See: https://api.slack.com/interactivity/slash-commands, and
-    https://api.slack.com/interactivity/handling#message_responses
+    Example URL: "http://localhost:9980/webhooks/<webhook_slug>?cmd=list_drafts"
 
-    In this sample, we expect the slash command's text to be:
-    - "gmail get profile"
-    - "gmail drafts list [optional query]"
-    - "gmail drafts get <draft ID>"
-    - "gmail messages list [optional query]"
-    - "gmail messages get <message ID>"
-    - "gmail messages send <short message to yourself>"
+    Commands:
+    - cmd=get_profile
+    - cmd=list_drafts&query=optional_query
+    - cmd=get_draft&draft_id=<draft_ID>
+    - cmd=list_messages&query=optional_query
+    - cmd=get_message&message_id=<message_ID>
+    - cmd=send_message&text=<message_text>
 
     Args:
-        event: Slack event data.
-    """
-    for cmd, handler in COMMANDS.items():
-        if event.data.text.startswith(cmd):
-            handler(event.data.user_id, event.data.text[len(cmd) + 1 :])
-            return
-
-
-def _get_profile(slack_channel, _):
-    """https://developers.google.com/resources/api-libraries/documentation/gmail/v1/python/latest/gmail_v1.users.html#getProfile
-
-    Args:
-        slack_channel: Slack channel name/ID to post debug messages to.
-        _: Unused suffix of the user's Slack command, if any.
+        event: HTTP request event data (contains query parameters).
     """
+    params = event.data.url.query
+    cmd = params.get("cmd")
+
+    if cmd == "get_profile":
+        _get_profile()
+    elif cmd == "list_drafts":
+        _drafts_list(params.get("query", ""))
+    elif cmd == "get_draft":
+        _drafts_get(params.get("draft_id"))
+    elif cmd == "list_messages":
+        _messages_list(params.get("query", ""))
+    elif cmd == "get_message":
+        _messages_get(params.get("message_id"))
+    elif cmd == "send_message":
+        _messages_send(params.get("text"))
+    else:
+        return "Unknown command"
+
+
+def _get_profile():
+    """https://developers.google.com/resources/api-libraries/documentation/gmail/v1/python/latest/gmail_v1.users.html#getProfile"""
     resp = gmail.getProfile(userId="me").execute()
-    slack.chat_postMessage(channel=slack_channel, text=resp["emailAddress"])
-    msg = f"Total no. of messages: `{resp['messagesTotal']}`"
-    slack.chat_postMessage(channel=slack_channel, text=msg)
-    msg = f"Total no. of threads: `{resp['threadsTotal']}`"
-    slack.chat_postMessage(channel=slack_channel, text=msg)
-    msg = f"Current History record ID: `{resp['historyId']}`"
-    slack.chat_postMessage(channel=slack_channel, text=msg)
+    print(resp["emailAddress"])
+    print(f"Total no. of messages: {resp['messagesTotal']}")
```

Comment:
In the case of these prints here, and error prints below, I think `print("xxx:", yyy)` is better than `print(f"xxx: {yyy}")`